source/module_base/matrix3.h

```diff
@@ -5,75 +5,273 @@
-Matrix3 operator -(const Matrix3 &m1, const Matrix3 &m2);	//m1-m2
-Matrix3 operator /(const Matrix3 &m,const double &s);		//m/s
-Matrix3 operator *(const Matrix3 &m1,const  Matrix3 &m2);	//m1*m2
-Matrix3 operator *(const Matrix3 &m,const double &s);		//m*s
-Matrix3 operator *(const double &s, const Matrix3 &m);		//s*m
-template<typename T> ModuleBase::Vector3<double> operator *(const Matrix3 &m, const ModuleBase::Vector3<T> &u);	//m*u				// Peize Lin change ModuleBase::Vector3<T> 2017-01-10
-template<typename T> ModuleBase::Vector3<double> operator *(const ModuleBase::Vector3<T> &u, const Matrix3 &m);	//u*m				// Peize Lin change ModuleBase::Vector3<T> 2017-01-10
+/**
+ * @brief Overload operator "+" for two 3x3 matrices m1 and m2
+ * i.e. m1+m2
+ *
+ * @param m1
+ * @param m2
+ * @return Matrix3
+ */
+Matrix3 operator+(const Matrix3 &m1, const Matrix3 &m2);
+
+/**
+ * @brief Overload operator "-" for two 3x3 matrices m1 and m2,
+ * i.e. m1-m2
+ *
+ * @param m1
+ * @param m2
+ * @return Matrix3
+ */
+Matrix3 operator-(const Matrix3 &m1, const Matrix3 &m2);
 
-bool operator ==(const Matrix3 &m1, const Matrix3 &m2); //whether m1 == m2
-bool operator !=(const Matrix3 &m1, const Matrix3 &m2); //whethor m1 != m2
+/**
+ * @brief Overload operator "/" for a (3x3 matrix)/(scalar)
+ * i.e. m/s
+ *
+ * @param m The 3x3 matrix
+ * @param s The scalar
+ * @return Matrix3
+ */
+Matrix3 operator/(const Matrix3 &m, const double &s);
 
+/**
+ * @brief Overload operator "*" for two 3x3 matrices m1 and m2
+ * i.e. m1*m2
+ *
+ * @param m1
+ * @param m2
+ * @return Matrix3
+ */
+Matrix3 operator*(const Matrix3 &m1, const Matrix3 &m2);
 
+/**
+ * @brief Overload operator "*" for (3x3 matrix)*(scalar)
+ * i.e. m*s
+ *
+ * @param m The 3x3 matrix
+ * @param s The scalar
+ * @return Matrix3
+ */
+Matrix3 operator*(const Matrix3 &m, const double &s);
 
+/**
+ * @brief Overload operator "*" for (scalar)*(3x3 matrix)
+ * i.e. s*m
+ *
+ * @param s The scalar
+ * @param m The 3x3 matrix
+ * @return Matrix3
+ */
+Matrix3 operator*(const double &s, const Matrix3 &m);
 
-//m*u
-template<typename T>
-ModuleBase::Vector3<double> operator *(const Matrix3 &m, const ModuleBase::Vector3<T> &u)
+/**
+ * @brief Over load operator "*" for (3x3 matrix)*(Vector3)
+ *
+ * @tparam T
+ * @param m The 3x3 matrix
+ * @param u The vector with 3 elements
+ * @return ModuleBase::Vector3<double>
+ * @author Peize Lin
+ */
+template <typename T> ModuleBase::Vector3<double> operator*(const Matrix3 &m, const ModuleBase::Vector3<T> &u);
+
+/**
+ * @brief Over load operator "*" for (Vector3)*(3x3 matrix)
+ *
+ * @tparam T
+ * @param u The vector with 3 elements
+ * @param m The 3x3 matrix
+ * @return ModuleBase::Vector3<double>
+ */
+template <typename T> ModuleBase::Vector3<double> operator*(const ModuleBase::Vector3<T> &u, const Matrix3 &m);
+
+/**
+ * @brief Overload operator "==" to assert
+ * the equality between two 3x3 matrices
+ *
+ * @param m1
+ * @param m2
+ * @return true
```

Comment:
The content following`@return` mark does not only mean the return type(e.g. "ModuleBase::Vector3<double>") or the return value ("true" or "false"); it should be a guideline for programmers to use the result appropriately. For example,
> @return true if these matrices are equal, otherwise false.

or,
> @return The sum of two Matrix3.

However, it is a little long-winded for simple cases like module_base.